From c66d505e5bb8e358468421e1f3679d919cffa21c Mon Sep 17 00:00:00 2001
From: Richard Gong <richard.gong@intel.com>
Date: Fri, 30 Apr 2021 08:36:21 -0500
Subject: [PATCH 27/41] HSD #22012990567-1: firmware: stratix10-svc: clean up
 SMC call

commit 2b9605976d8aac38c4471b3851bceb0382e770e4 from
https://github.com/altera-opensource/linux-socfpga.git
branch is socfpga-5.4.114-lts

Clean up to use the combined SMC call function.

Signed-off-by: Richard Gong <richard.gong@intel.com>
Integrated-by: Abhishek Paliwal <paliwal.abhishek@windriver.com>
---
 drivers/firmware/stratix10-svc.c | 17 +++++++----------
 1 file changed, 7 insertions(+), 10 deletions(-)

diff --git a/drivers/firmware/stratix10-svc.c b/drivers/firmware/stratix10-svc.c
index 02cb5acd3201..8ee1815c08d0 100644
--- a/drivers/firmware/stratix10-svc.c
+++ b/drivers/firmware/stratix10-svc.c
@@ -383,7 +383,7 @@ static int svc_normal_to_secure_thread(void *data)
 	struct stratix10_svc_data *pdata;
 	struct stratix10_svc_cb_data *cbdata;
 	struct arm_smccc_res res;
-	unsigned long a0, a1, a2, a3, a4, a5;
+	unsigned long a0, a1, a2, a3, a4, a5, a6, a7;
 	int ret_fifo = 0;
 
 	pdata =  kmalloc(sizeof(*pdata), GFP_KERNEL);
@@ -403,6 +403,8 @@ static int svc_normal_to_secure_thread(void *data)
 	a3 = 0;
 	a4 = 0;
 	a5 = 0;
+	a6 = 0;
+	a7 = 0;
 
 	pr_debug("smc_hvc_shm_thread is running\n");
 
@@ -532,15 +534,10 @@ static int svc_normal_to_secure_thread(void *data)
 		pr_debug("%s: before SMC call -- a0=0x%016x a1=0x%016x",
 			 __func__, (unsigned int)a0, (unsigned int)a1);
 		pr_debug(" a2=0x%016x\n", (unsigned int)a2);
-
-		if (a0 == INTEL_SIP_SMC_FCS_CRYPTION) {
-			pr_debug(" a3=0x%016x\n", (unsigned int)a3);
-			pr_debug(" a4=0x%016x\n", (unsigned int)a4);
-			pr_debug(" a5=0x%016x\n", (unsigned int)a5);
-			ctrl->invoke_fn(a0, a1, a2, a3, a4, a5, 0, 0, &res);
-		} else {
-			ctrl->invoke_fn(a0, a1, a2, 0, 0, 0, 0, 0, &res);
-		}
+		pr_debug(" a3=0x%016x\n", (unsigned int)a3);
+		pr_debug(" a4=0x%016x\n", (unsigned int)a4);
+		pr_debug(" a5=0x%016x\n", (unsigned int)a5);
+		ctrl->invoke_fn(a0, a1, a2, a3, a4, a5, a6, a7, &res);
 
 		pr_debug("%s: after SMC call -- res.a0=0x%016x",
 			 __func__, (unsigned int)res.a0);
-- 
2.18.1

